crates/oxc_minifier/src/peephole/minimize_statements.rs

```diff
@@ -0,0 +1,380 @@
+            result.push(stmt);
+        }
+
+        // Drop a trailing unconditional jump statement if applicable
+        if let Some(last_stmt) = result.last() {
+            match last_stmt {
+                // "while (x) { y(); continue; }" => "while (x) { y(); }"
+                Statement::ContinueStatement(s) if s.label.is_none() => {
+                    if let Some(Ancestor::ForStatementBody(_)) = ctx.ancestors().nth(1) {
+                        result.pop();
+                        self.mark_current_function_as_changed();
+                    }
+                }
+                // "function f() { x(); return; }" => "function f() { x(); }"
+                Statement::ReturnStatement(s) if s.argument.is_none() => {
+                    if let Ancestor::FunctionBodyStatements(_) = ctx.parent() {
+                        result.pop();
+                        self.mark_current_function_as_changed();
+                    }
+                }
+                _ => {}
+            }
+        }
+
+        // Merge certain statements in reverse order
+        if result.len() >= 2 {
+            if let Some(Statement::ReturnStatement(_)) = result.last() {
+                'return_loop: while result.len() >= 2 {
+                    let prev_index = result.len() - 2;
+                    let prev_stmt = &result[prev_index];
+                    match prev_stmt {
+                        Statement::ExpressionStatement(_) => {
+                            if let Some(Statement::ReturnStatement(last_return)) = result.last() {
+                                if last_return.argument.is_none() {
+                                    break 'return_loop;
+                                }
+                            }
+                            self.mark_current_function_as_changed();
+                            // "a(); return b;" => "return a(), b;"
+                            let last_stmt = result.pop().unwrap();
+                            let Statement::ReturnStatement(mut last_return) = last_stmt else {
+                                unreachable!()
+                            };
+                            let prev_stmt = result.pop().unwrap();
+                            let Statement::ExpressionStatement(mut expr_stmt) = prev_stmt else {
+                                unreachable!()
+                            };
+                            let b = last_return.argument.as_mut().unwrap();
+                            let argument = Self::join_sequence(&mut expr_stmt.expression, b, ctx);
+                            let right_span = last_return.span;
+                            let last_return_stmt =
+                                ctx.ast.statement_return(right_span, Some(argument));
+                            result.push(last_return_stmt);
+                        }
+                        // Merge the last two statements
+                        Statement::IfStatement(if_stmt) => {
```

Comment:
Similar to the previous question, I wonder if this can be done inside `self.minimize_statement` by running `self.minimize_statement` when the last item of `result` changed.